### PR TITLE
Arg parser

### PR DESCRIPTION
**Pull request by joojis**

### CODE REVIEW

**Comment by joojis** [comment]
For GitTask or any other Task that requires List<String> args on execute function, I have concern if the case exists such that we need to pass options/flags to the List<String> args. Bypassing options/flags are not implemented in ArgParser yet.(https://github.com/dart-lang/args/issues/9)

File: beaver_task/lib/src/task/delete_task.dart [hunk 1]
```diff
@@ -22,7 +23,14 @@ class DeleteTask extends Task {
       : force = force,
         recursive = recursive;
 
-  DeleteTask.fromArgs(List<String> args) : this(args);
+  factory DeleteTask.fromArgs(List<String> args) {
+    final parser = new ArgParser();
+    parser.addFlag('force', defaultsTo: true, abbr: 'f');
```
Comment (kseo) [inline beaver_task/lib/src/task/delete_task.dart]: Use method cascades.

File: beaver_task/lib/src/task/git_task.dart [hunk 2]
```diff
@@ -14,7 +15,13 @@ class GitTask extends Task {
   GitTask(this.args, {String processWorkingDir})
       : this.processWorkingDir = processWorkingDir;
 
-  GitTask.fromArgs(List<String> args) : this(args);
+  factory GitTask.fromArgs(List<String> args) {
+    final parser = new ArgParser();
+    parser.addOption('process-working-dir', abbr: 'C');
```
Comment (kseo) [inline beaver_task/lib/src/task/git_task.dart]: Use method cascades.

`new ArgParser()..addOption('process-working-dir', abbr: 'C');`

File: beaver_task/lib/src/task/touch_task.dart [hunk 3]
```diff
@@ -17,7 +18,12 @@ class TouchTask extends Task {
 
   TouchTask(this.paths, {create: true}) : create = create;
 
-  TouchTask.fromArgs(List<String> args) : this(args);
+  factory TouchTask.fromArgs(List<String> args) {
+    final parser = new ArgParser();
+    parser.addFlag('create', defaultsTo: true);
```
Comment (kseo) [inline beaver_task/lib/src/task/touch_task.dart]: Use method cascades.

File: beaver_task/test/task/delete_task_test.dart [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import "package:test/test.dart";
+import "package:beaver_task/beaver_task.dart";
+
+void main() {
+  test("DeleteTask should be able to constructed from JSON string.", () {
```
Comment (kseo) [inline beaver_task/test/task/delete_task_test.dart]: to constructed -> to be constructed

File: beaver_task/test/task/touch_task_test.dart [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import "package:test/test.dart";
+import "package:beaver_task/beaver_task.dart";
+
+void main() {
+  test("TouchTask should be able to constructed from JSON string.", () {
```
Comment (kseo) [inline beaver_task/test/task/touch_task_test.dart]: to constructed -> to be constructed

**Pull request merged**